Remove cuda11 reference from CMake file

ChuckHastings opened the pull request:
Old CUDA 11 reference, RAPIDS no longer supports anything related to CUDA 11.

Fixes #5158

bdice commented:
This PR does not close #5158 as there are still several references to CUDA 11. Please evaluate the following and decide if changes are needed.

- https://github.com/rapidsai/cugraph/blob/2c957142ebfbe6c2220ec13e7d48142312001f93/cpp/tests/utilities/validation_utilities.cu#L423
- https://github.com/rapidsai/cugraph/blob/2c957142ebfbe6c2220ec13e7d48142312001f93/benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile#L16
- A bunch more in the `notebooks/` directory: https://github.com/search?q=repo%3Arapidsai%2Fcugraph%20%2Fcuda.*11%2F&type=code

bdice commented on python/cugraph/cugraph/community/ktruss_subgraph.py:
```diff
@@ -65,8 +65,6 @@ def ktruss_subgraph(
     """
     Returns the K-Truss subgraph of a graph for a specific k.
 
-    NOTE: this function is currently not available on CUDA 11.4 systems.
```
This is a "special" note, as it triggers the `skip_docstring` function.

This is the only remaining CUDA incompatibility that we mark in this way. Can we remove `skip_docstring` entirely?

https://github.com/rapidsai/cugraph/blob/2c957142ebfbe6c2220ec13e7d48142312001f93/python/cugraph/cugraph/tests/docs/test_doctests.py#L122

BradReesWork replied:
There are actually 4 PRs against the one CUDA 11 issues.  I just put the close in this PR that was planned to be the last

ChuckHastings replied:
Merged @bdice's change from #5177 which updates skip_docstring to remove the version logic.  We can always add it back later if we have a case for that... but hopefully we're learning not to do that :-)

jakirkham commented:
Thanks Chuck! 🙏

Should we update this CUDA 11 reference as well?

https://github.com/rapidsai/cugraph/blob/068c441216cec2562bd0f2948a41ece9488cda55/cpp/libcugraph_etl/CMakeLists.txt#L30-L33

ChuckHastings commented:
> Thanks Chuck! 🙏
> 
> Should we update this CUDA 11 reference as well?
> 
> https://github.com/rapidsai/cugraph/blob/068c441216cec2562bd0f2948a41ece9488cda55/cpp/libcugraph_etl/CMakeLists.txt#L30-L33

Yes.  @bdice made the same suggestion above.  I was out for a few days and just saw both comments.  Will get the change out today.

jakirkham commented:
Thanks Chuck! 🙏

Also noticing some notebooks reference CUDA 11 (and may not have been rerun recently). For example

https://github.com/rapidsai/cugraph/blob/branch-25.08/notebooks/demo/mg_property_graph.ipynb?short_path=d2f0f45#L14-L17

How would we like to handle these?

ChuckHastings commented:
> This PR does not close #5158 as there are still several references to CUDA 11. Please evaluate the following and decide if changes are needed.
> 
> * https://github.com/rapidsai/cugraph/blob/2c957142ebfbe6c2220ec13e7d48142312001f93/cpp/tests/utilities/validation_utilities.cu#L423

Fixed in the PR now.

> * https://github.com/rapidsai/cugraph/blob/2c957142ebfbe6c2220ec13e7d48142312001f93/benchmarks/shared/build_cugraph_ucx/cugraph_ucx.dockerfile#L16

Fixed in #5174 

> * A bunch more in the `notebooks/` directory: https://github.com/search?q=repo%3Arapidsai%2Fcugraph%20%2Fcuda.*11%2F&type=code

Still pondering.

jakirkham reviewed (approved):
LGTM

Thanks Chuck! 🙏

ChuckHastings commented:
/merge
